tools/rpkg/src/statement.cpp

```diff
@@ -268,20 +274,67 @@ static void transform(Vector &src_vec, SEXP &dest, idx_t dest_offset, idx_t n) {
 	case LogicalTypeId::INTEGER:
 		VectorToR<int32_t, uint32_t>(src_vec, n, INTEGER_POINTER(dest), dest_offset, NA_INTEGER);
 		break;
+	case LogicalTypeId::TIMESTAMP_SEC: {
+		auto src_data = FlatVector::GetData<int64_t>(src_vec);
```

Comment:
This code seems to be duplicated four times, could we extract this into a templated function?

For example, see how this is done [in the Python wrapper](https://github.com/duckdb/duckdb/blob/b48ff00354c88ea9ce4c9554adfa10689146cdd2/tools/pythonpkg/src/array_wrapper.cpp#L664).